src/ApplicationManager.cpp

```
@@ -965,6 +962,33 @@ void ApplicationManager::handleSpoolmanSynced(const AppMessage& msg) {
+                if (weightG != trayDashboardState_.trays[i].weight_g) {
+                    trayDashboardState_.trays[i].weight_g = weightG;
+                    Serial.printf("ApplicationManager: Dashboard tray %d weight updated to %dg\n",
+                                  trayDashboardState_.trays[i].tray_index, weightG);
+#ifndef NATIVE_TEST
+                    Preferences prefs;
+                    prefs.begin("spoolsense", false);
+                    prefs.putBytes("tray_dash", &trayDashboardState_, sizeof(TrayDashboardState));
+                    prefs.end();
+
+                    bool dashEnabled = ConfigurationManager::getInstance().isBambuDashboardEnabled();
+                    if (dashEnabled && display_) {
```

Clang (14.0.6)

[note] +5, including nesting penalty of 4, nesting level increased to 5

(clang)